homeassistant/components/synology_dsm/camera.py

```diff
@@ -73,7 +113,7 @@ def device_info(self) -> Dict[str, any]:
     @property
     def available(self) -> bool:
         """Return the availability of the camera."""
-        return self._camera.is_enabled
+        return self.coordinator.data[self._camera_id].is_enabled
 
```

Comment:
This can be deleted since its provided by `CoordinatorEntity`

We shouldn't be setting up the entity in `async_setup_entry` if its not enabled

Reply:
now, when camera is not enabled during `async_setup_entry`, it is still added, but entity remains disabled.
If camera is disabled during lifetime, `available` will switch to False.

The by `CoordinatorEntity`  provided `def available` does only respect `coordinator.last_update_success`, but not real state from `camera.is_enabled`, therefore i added:
```python
    @property
    def available(self) -> bool:
        """Return the availability of the camera."""
        return self.camera_data.is_enabled and self.coordinator.last_update_success
```